engine/src/test/java/io/camunda/zeebe/engine/processing/resource/ResourceDeletionTest.java

```diff
@@ -185,6 +188,79 @@ public void shouldEvaluateLatestVersionIfPreviousVersionIsDeleted() {
         .containsOnly("jedi_or_sith", 2, drgKeyV2);
   }
 
+  @Test
+  public void shouldHaveCorrectLifecycleWhenDeletingProcessWithoutRunningInstances() {
+    // given
+    final var processId = helper.getBpmnProcessId();
+    final var processDefinitionKey = deployProcess(processId);
+
+    // when
+    engine.resourceDeletion().withResourceKey(processDefinitionKey).delete();
+
+    // then
+    assertThat(
+            RecordingExporter.records()
+                .onlyEvents()
+                .limit(r -> r.getIntent().equals(ResourceDeletionIntent.DELETED)))
+        .describedAs("Should write events in correct order")
+        .extracting(Record::getIntent)
+        .containsExactly(
+            ProcessIntent.CREATED,
+            DeploymentIntent.CREATED,
+            ResourceDeletionIntent.DELETING,
+            ProcessIntent.DELETING,
+            ProcessIntent.DELETED,
+            ResourceDeletionIntent.DELETED);
+  }
+
+  @Test
+  public void shouldWriteEventsForDeletedProcessWithoutRunningInstances() {
+    // given
+    final var processId = helper.getBpmnProcessId();
+    final var processDefinitionKey = deployProcess(processId);
+
+    // when
+    engine.resourceDeletion().withResourceKey(processDefinitionKey).delete();
+
+    // then
+    verifyProcessIsDeleted(processId, 1);
+    verifyResourceIsDeleted(processDefinitionKey);
+  }
+
+  @Test
+  public void shouldCreateInstanceOfVersionOneWhenVersionTwoIsDeleted() {
+    // given
+    final var processId = helper.getBpmnProcessId();
+    deployProcess(processId);
+    final var secondProcessDefinitionKey = deployProcess(processId);
+    engine.resourceDeletion().withResourceKey(secondProcessDefinitionKey).delete();
+
+    // when
+    final var processInstanceKey = engine.processInstance().ofBpmnProcessId(processId).create();
+
+    // then
+    verifyProcessIsDeleted(processId, 2);
+    verifyResourceIsDeleted(secondProcessDefinitionKey);
+    verifyProcessInstanceIsCompleted(processId, 1, processInstanceKey);
```

Comment:
🔧 For readability of these tests it could help to add something about version to the method name. Currently, it's not clear what the magic numbers are. For example,

```java
verifyProcessIdAndVersionIsDeleted(processId, 2);
..
verifyInstanceOfProcessWithIdAndVersion(processInstanceKey, processId, 1);
```

Alternatively, you could replace `verifyProcessInstanceIsCompleted` to something like `verifyProcessIdAndVersionOfProcessInstance(processInstanceKey, processId, 1)`

Reply:
I guess I'm spoiled by IntelliJ putting down the parameter names so much that I don't care about magic numbers anymore 😛 I'll change the methods names to clarify.